bio/bioconvert/wrapper.py

```
@@ -0,0 +1,53 @@
+        extra += f" --threads {snakemake.threads}"
+
+    commands_expecting_reference = {
+        "bam2cram",
+        "cram2bam",
+        "cram2fasta",
+        "cram2fastq",
+        "cram2sam",
+        "sam2cram",
+    }
+    if converter in commands_expecting_reference:
+        extra += f" --reference '{snakemake.input.ref}'"
```

Ruff (0.15.21)

[error] Undefined name `snakemake`

(F821)